src/inmanta/ast/statements/assign.py

```diff
@@ -608,17 +608,24 @@ def get_field(self, key: str, args: abc.Sequence[object], kwds: abc.Mapping[str,
         Overrides Formatter.get_field. Composite variable names are expected to be resolved at this point and can be
         retrieved by their full name.
         """
-        return (kwds[key], key)
+        return (kwds[key.strip()], key)
 
 
 class StringFormatV2(FormattedString):
     """
     Create a new string by using python build in formatting
+
     """
 
     __slots__ = ()
 
     def __init__(self, format_string: str, variables: abc.Sequence[typing.Tuple["Reference", str]]) -> None:
+        """
+        :param format_string: The string on which to perform substitution
+        :param variables: Sequence of tuples each holding a reference to a variable to substitute in the format_string
```

Comment:
This is still not clear to me:

1. what is responsible for removing whitespaces? (i.e. where can we expect them to exist?)
2. I would avoid trying to re-construct the matched strings (which we do here: `full_name` is built up in 'AttributeReference` , so any kind of garbage in there makes this fail with a weird error)


I would actually use the pair we get here to construct a lookup between `literal value with all kinds of garbage` and `clean reference` and I would strip all whitespace of all individual parts of the reference.